.github/workflows/ci.yml

```diff
@@ -0,0 +1,29 @@
+name: MacOS application
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: macOS-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: "NeedleFoundationTests"
```

Comment:
I don't think they need to be one word, you can add spaces or make them a sentence to make it easier to understand.